ext/standard/http_fopen_wrapper.c

```diff
@@ -203,6 +203,13 @@ static php_stream *php_stream_url_wrap_http_ex(php_stream_wrapper *wrapper,
 
 	if (context && (tmpzval = php_stream_context_get_option(context, wrapper->wops->label, "timeout")) != NULL) {
 		double d = zval_get_double(tmpzval);
+
+		if (d > (double) PHP_TIMEOUT_ULL_MAX / 1000000.0) {
```

Comment:
Using `PHP_TIMEOUT_ULL_MAX` here (and maybe elsewhere) is presumptious. While that is likely good for POSIX platforms ("time_t shall be an integer type with a width (see [<stdint.h>](https://pubs.opengroup.org/onlinepubs/9799919799/basedefs/stdint.h.html) ) of at least 64 bits") (albeit not perfect since `time_t` could be signed), it's completely borked for Windows where `long` is `uint32_t` for x64 and x86. So for Windows, that needs to be:
````c
if (d > (double) LONG_MAX / 1000000.0) {
````
For stable branches, maybe define a `const` variable conditionally (`#ifndef PHP_WIN32`), and use that for the check.

Reply:
> it's completely borked for Windows where long is uint32_t for x64 and x86

Oh.